Add support and documentation for Windows

XavierCooney opened the pull request:
After quite a lot of trial and error, I managed to get Fabrik running natively on Windows. I've written out my process. One of the changes I needed to make was to make Fabrik compatible with Python 3, as Tensorflow for Windows can only run on Python 3.

I've probably made some mistakes on the way, and I'm still working on documenting the Visual C++ installation step, so this is more of a semi WIP, but is pretty much done.

thatbrguy commented:
@XavierCooney Hello, that's some impressive work! Few things before we approve your task: 

- Is it still semi-WIP? 
- Provide a short gif or video demonstrating launching Fabrik on windows. In the video,  run `celery -A ide worker --app=ide.celery_app  --loglevel=info` and `python manage.py runserver` to launch Fabrik. Then, load a model from the zoo, and then export it (any framework). Clear the workspace and import the model from the file you saved earlier. 
- Share the above video/gif to our [gitter channel](https://gitter.im/Cloud-CV/Fabrik)

thatbrguy commented on WINDOWS.md:
```diff
@@ -0,0 +1,144 @@
+
+### WSL Notes
+
+The setup is exactly the same as other linux systems. The Ubtuntu disto has been tested and confirmed to work.
```
Mention ubuntu 14.04 or 16.04

thatbrguy commented on WINDOWS.md:
```diff
@@ -0,0 +1,144 @@
+### WSL Notes
+
+The setup is exactly the same as other linux systems. The Ubtuntu disto has been tested and confirmed to work.
+It is iportant Fabrik is setup in `/mnt/c/...`, rather than the Linux user profile directory, because editing of the WSL subsystem from the Win32 subsystem can leading to file corruption. 
```
```suggestion
It is important that Fabrik is setup in `/mnt/c/...`, rather than the Linux user profile directory, because editing of the WSL subsystem from the Win32 subsystem can leading to file corruption. 
```
Minor typo

thatbrguy commented on WINDOWS.md:
```diff
@@ -0,0 +1,144 @@
+
+5. Install Tensorflow, Keras and Caffe. Note that this will fail if the wrong version of Python is used.
+    
+    Tenserflow (this will fail if you don't have either python 3.6, 3.5 or 3.4 64 bit):
```
```suggestion
- Tensorflow (this will fail if you don't have either python 3.6, 3.5 or 3.4 64 bit):
```
Minor typo

thatbrguy commented on WINDOWS.md:
````diff
@@ -0,0 +1,144 @@
+    ```
+    (venv) cmd> pip install tensorflow==1.4
+    ```
+    Keras:
````
```suggestion
- Keras:
```

thatbrguy commented on WINDOWS.md:
````diff
@@ -0,0 +1,144 @@
+    ```
+    (venv) cmd> pip install keras==2.0.8
+    ```
+    A work-in-progess Windows port of Caffe is currently maintaiend 
````
```suggestion
- A work-in-progess Windows port of Caffe is currently maintaiend 
```

thatbrguy commented on WINDOWS.md:
```diff
@@ -0,0 +1,144 @@
+
+
+6. Install other dependencies. Edit `requirements\common.txt` line 8 and change `tensorflow==1.4.1` to `tensorflow==1.4`.
+    Comment out the `uWSGI` line, as it is incompatible with windows and is only needed for depoyment.
```
```suggestion
    Comment out the `uWSGI` line, as it is incompatible with windows and is only needed for deployment.
```
Minor typo

thatbrguy commented on WINDOWS.md:
```diff
@@ -0,0 +1,144 @@
+
+6. Install other dependencies. Edit `requirements\common.txt` line 8 and change `tensorflow==1.4.1` to `tensorflow==1.4`.
+    Comment out the `uWSGI` line, as it is incompatible with windows and is only needed for depoyment.
+    Install Microsoft Visual C++ Tools for the Twisted package by installing following https://visualstudio.microsoft.com/visual-cpp-build-tools/. Install
```
```suggestion
    Install Microsoft Visual C++ Tools for the Twisted package by installing the [following](https://visualstudio.microsoft.com/visual-cpp-build-tools/). Install
```
Adding markdown

thatbrguy commented:
@XavierCooney Hello, please address the review suggestions in addition to the tasks I mentioned earlier.

XavierCooney commented:
@thatbrguy Hi, I've fixed the typos, and I realised that some of the tests were failing on windows because of python 2-3 incompatibility, so I fixed them too. I'll get the GIF recorded soon. Thanks.

Ram81 commented:
@XavierCooney can you reopen the same PR on `windows-setup` branch

XavierCooney commented:
Closed in favour of #415.
